tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,300 @@
+typedef struct nns_src *nns_src_h;
+
+/**
+ * @brief A handle of a "switch" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_switch *nns_switch_h;
+
+/**
+ * @brief A handle of a "valve node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_valve *nns_valve_h;
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer Pipelines.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_ERROR_NONE				= TIZEN_ERROR_NONE, /**< Success! */
+  NNS_ERROR_INVALID_PARAMETER			= TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  NNS_ERROR_NOT_SUPPORTED			= TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_NO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline state
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI) Fill with GST's state values.
+ * @todo Check the consistency against MMFW APIs.
+ */
+typedef enum {
+  NNS_PIPELINE_UNKNOWN				= 0, /**< Unknown state. Maybe not constructed? */
+  NNS_PIPELINE_NULL,				= 1, /**< GST-State "Null" */
+  NNS_PIPELINE_READY,				= 2, /**< GST-State "Ready" */
+  NNS_PIPELINE_PAUSED,				= 3, /**< GST-State "Paused" */
+  NNS_PIPELINE_PLAYING,				= 4, /**< GST-State "Playing" */
+} nns_pipeline_state;
+
+/**
+ * @brief Enumeration for switch types
+ * @detail This designates different GStreamer filters, "GstInputSelector"/"GetOutputSelector".
+ * @since_tizen 5.5
+ * @todo There may be more filters that can be supported.
+ */
+typedef enum {
+  NNS_SWITCH_OUTPUTSELECTOR,			= 0, /** GstOutputSelector */
+  NNS_SWITCH_INPUTSELECTOR,			= 1, /** GstInputSelector */
+} nns_switch_type;
+
+
+/****************************************************
+ ** NNStreamer Pipeline Construction (gst-parse)   **
+ ****************************************************/
+/**
+ * @brief Construct the pipeline (GStreamer + NNStreamer)
+ * @details Use this function to create a gst_parse_launch compatible NNStreamer pipelines.
+ * @since_tizen 5.5
+ * @param[in] pipeline_description The pipeline description compatible with GStreamer gst_parse_launch(). Refer to GStreamer manual or NNStreamer (github.com/nnsuite/nnstreamer) documentation for examples and the grammar.
+ * @param[out] pipe The nnstreamer pipeline handler from the given description
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_construct (const char *pipeline_description, nns_pipeline_h *pipe);
```

Comment:
can these return types be enum as well, possible nns_error_e or a new enum?

Reply:
According to other Tizen C APIs, it appears that it is supposed to use int there:

https://developer.tizen.org/development/api-references/native-application?redirect=https://developer.tizen.org/dev-guide/5.0.0/org.tizen.native.mobile.apireference/group__CAPI__MEDIA__STREAMER__MODULE.html

Reply:
Yes.